core/Widget/WidgetsList.php

```diff
@@ -139,7 +139,15 @@ public function addToContainerWidget($containerId, WidgetConfig $widget)
     public function remove($widgetCategoryId, $widgetName = false)
     {
         foreach ($this->widgets as $index => $widget) {
-            if ($widget->getCategoryId() === $widgetCategoryId) {
+            if (
+                ($widget->getCategoryId() === $widgetCategoryId)
+                ||
+                (
+                    (substr_compare($widgetCategoryId, '*', -1) === 0)
+                    &&
+                    (substr_compare($widget->getCategoryId(), $widgetCategoryId, 0, strlen($widgetCategoryId) - 1) === 0)
+                )
+            ) {
```

Comment:
I can revert this and remove all the promo widgets one by one, or keep this and add a test for this and update the docblock.